mindsdb/integrations/handlers/ray_serve_handler/ray_serve_handler.py

```diff
@@ -59,14 +71,29 @@ def predict(self, df, args=None):
         args = {**(self.model_storage.json_get('args')), **args}  # merge incoming args
         pred_args = args.get('predict_params', {})
         args = {**args, **pred_args}  # merge pred_args
-        resp = requests.post(args['predict_url'],
-                             json={'df': df.to_json(orient='records'), 'pred_args': pred_args},
-                             headers={'content-type': 'application/json; format=pandas-records'})
-
+        if args.get('parquet', False):
+            buffer = io.BytesIO()
+            df.attrs['pred_args'] = pred_args
+            df.to_parquet(buffer)
+            buffer.seek(0)
+            parquet_data = buffer.read()
+            headers = {'Content-Type': 'application/octet-stream'}
+            resp = requests.post(args['predict_url'], data=parquet_data, headers=headers)
+        else:
+            resp = requests.post(args['predict_url'],
+                                 json={'df': df.to_json(orient='records'), 'pred_args': pred_args},
+                                 headers={'content-type': 'application/json; format=pandas-records'})
         try:
-            response = resp.json()
+            if args.get('parquet', False):
```

Comment:
Why do we need parquet data for the prediction?

Reply:
I think if ray service is using parquet as input it should use the same format as output. so it is ok to have same formats

Reply:
Yeah, but ideally, we would want to define this in the predict query, so different services can query the model and receive the format they want to use.